### ISSUE TITLE
Wrapped text shifts left when company auto-complete comes up

### ISSUE THREAD

**Issue by codygman** [rnkn/olivetti#53]
emacs: 28.0.50
olivetti version:

;; Package-Version: 20200702.601
;; Version: 1.11.1

Given text like this and a configuration where olivetti wraps the line:

```
complete after this: wo_ <---- cursor here, then trigger completion window
* This is a very very very long line that is longer than my olivetti mode keeps on one line and wraps to the second
```

That second line will shift to the left and it's very visually distracting lately, though I could easily ignore it at first.

Related:

company-mode/company-mode#272
zk-phi/sublimity#21

**Comment by rnkn** [comment]
Can't reproduce this sorry...
<img width="1272" alt="Screen Shot 2020-08-24 at 4 34 38 pm" src="https://user-images.githubusercontent.com/1256849/91011500-c0bdd200-e627-11ea-88d3-9610e635f524.png">

**Comment by dgutov** [comment]
Perhaps you have to use the same version of company to reproduce (most recent, from MELPA, right?).

**Comment by rnkn** [comment]
Then that would indicate a regression in company-mode.

**Comment by codygman** [comment]
I'll try to post exact versions and a screencast gif with repro later.

**Comment by rnkn** [comment]
The other possibility is that someone broke the window display code in Emacs again, given that this happened three times in the last four major versions...

If you can provide a backtrace that would be more informative than versions. Maybe also have a look in your NEWS file for relevant changes

**Comment by dgutov** [comment]
I haven't tried olivetti yet, but the relevant change in Company might be switching to using `after-string` instead of `display` overlay property for popup rendering. Which will be needed to show icons in the popup. Perhaps it interferes with stuff this mode modes?

**Comment by rnkn** [comment]
Nah, all this code does is essentially call set-window-margins on window-size-change-functions hook. There's no code here that could do what's being reported.

I've reopened on the off-chance that there's a rather elaborate regression in Emacs master that might change the way margins are set, but probably not.

If I had to guess I'd say that the company-mode change from display prop to after-string could be messing up Emacs's ability to accurately count the window width columns, but since I can't reproduce it (with current company-mode) it's just a guess.
